[ML] Render Custom URL as textarea on focus

darnautov opened the pull request:
## Summary

Improve the Custom URL form usability by rendering focused URL input as textarea.  
![image](https://user-images.githubusercontent.com/5236598/66741201-3fc57400-ee75-11e9-99b2-27c3737060ff.png)

### Checklist

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- [ ] ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [x] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

darnautov committed:
[ML] custom url as textarea on focus

darnautov committed:
[ML] update snapshot

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

darnautov commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into ML-custom-urls-inputs

cauemarcondes committed:
[APM] Metadata sections should show message when no data is available (#47842)

* Show message if section is required or if property does not have value and is required

* Filtering out sections not required without properties value

* Filtering out sections not required without properties value

* improving test description

* removing unused import

patrykkopycinski committed:
[SIEM] Add tabbed sub-navigation to Network page (#47556)

ChrsMark committed:
Add Coredns Dashboard and logo (#47961)

peteharverson committed:
[ML] Fixes anomalies table link to view examples (#47954)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/4233/)
* Commit: c3186657090894649f8978718ab5404e1cb45b5d

alexwizp committed:
Prepare index_patterns/index_pattern.ts for move - remove ui/notify (#47969)

azasypkin committed:
Drop `/api/security/v1/saml` route in favour of `/api/security/saml/callback`. (#47929)

rudolf committed:
Remove platform as code owner from ui/saved_objects (#47673)

walterra reviewed (approved):
LGTM

jgowdyelastic committed:
[ML] Fixing wizard inputs functional tests (#48076)

Commit:
[code] Make go langserver use the bundled git (#48047)

* [code] Make go langserver use the bundled git

Go langserver needs vcs tools, like git, to resolve and download the
dependencies. However there is no gurantee that client has the git
with proper version. Add the bundled git into the $PATH when we
spawn go langserver process.

* [code] Pass the git related env vars when spwan go langserver.

In order to make the bundled git work correctly, pass some env variables
when code server spwan go langserver.

The env var `GIT_TEMPLATE_DIR` and `GIT_EXEC_PATH` is not necessary for
the git used by golang.

flash1293 committed:
fix wording in field editor (#48068)

flash1293 committed:
Saved object finder: Include freely specifable fields in request (#47321)

peteharverson commented on x-pack/legacy/plugins/ml/public/jobs/components/custom_url_editor/list.js:
```diff
@@ -170,15 +172,24 @@ export const CustomUrlList = injectI18n(class CustomUrlList extends Component {
+                <EuiTextArea
+                  value={customUrl.url_value}
+                  onChange={(e) => this.onUrlValueChange(e, index)}
+                  onBlur={() => this.setState({ expandedUrlIndex: null })}
```
I've noticed that the input doesn't collapse back to a text field if you mouse click, for example, in the 'Label' field of a different custom URL. Is there anything that can be done about that, so that the field is only a text area when it has focus? Seems to work mostly as intended on tab navigation.

darnautov committed:
[ML] refactor to TS, fix blur behaviour

darnautov committed:
Merge remote-tracking branch 'upstream/master' into ML-custom-urls-inputs

jgowdyelastic reviewed (approved):
LGTM

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/4313/)
* Commit: b0200875c996739cdd5382ddd9bc8d2c41cada15

peteharverson reviewed (approved):
Tested latest edits, and LGTM

darnautov commented:
retest

darnautov committed:
[ML] update tests for custom urls list

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/4410/)
* Commit: 3072f61b95f35df696cad9d707235b270b17aa73
